src/System.Private.CoreLib/src/Internal/Threading/Tasks/AsyncCausalitySupport.cs

```diff
@@ -57,6 +57,18 @@ public static void TraceOperationCompletedError(Task task)
         {
             AsyncCausalityTracer.TraceOperationCompletion(CausalityTraceLevel.Required, task.Id, AsyncCausalityStatus.Error);
         }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static void TraceSynchronousWorkStart(Task task)
```

Comment:
This type is meant to have just enough async debugger support needed by the AsTask() extension methods in the WindowsRuntimeSystemExtensions class. (See comment at the top.)

I do not think we want to be adding more public methods to it.

Reply:
Oops, I didn't realize it is public.

Reply:
I changed the two new methods to be internal. If that's not a good solution I suggest to revert https://github.com/dotnet/corert/pull/6832 and go back to the drawing board. I can probably reshuffle the dependencies in a way that will make it cleaner.

Reply:
This is ok interim solution. The comment at the top of this file is not correct anymore - it should be update accordingly in next iteration if these methods stay.